napari_skimage_regionprops/_regionprops.py

```diff
@@ -184,3 +196,371 @@ def ellipsoid_axis_lengths(table):
+    summary: bool, optional
+        Determine whether to return summary statistics or direct relatioships.
+        Default is True.
+    suffixes: List[str], optional
+        Suffixes to identify each channel. List must match number of channel in
+        'labels_array'. If None, then '-ch' + number is used.
+
+    Returns
+    -------
+    table_list : List
+        List of tables where each table relates one probe channel objects to
+        the reference channel objects. If `summary` is True, for each property,
+        returns count, mean, standard deviation, median, min, max, 25th
+        quartile and 75th quartile. If `summary` is False, direct relationships
+        are returned (which may lead to columns with repeated labels).
+    """
+    # quantitative analysis using scikit-image's regionprops
+    from skimage.measure import regionprops_table as sk_regionprops_table
+    import numpy as np
+    import pandas as pd
+    # To DO: check if input shape is correct
+    table_list = []
+    # Single channel case (single label image provided)
+    if ref_channel is None:
+        table_list += [regionprops_table(image = intensity_image,
+                                        labels = labels_array,
+                                        **kwargs)]
+        return table_list
+    # Channel axis is expected to be 0
+    n_channels = labels_array.shape[0]
+   
+    large_numbers = False
+    # If not provided, create default suffixes
+    if not suffixes:
+        suffixes = ['-ch' + str(i) for i in range(n_channels)]
+    else:
+        suffixes = ['-' + suffix for suffix in suffixes]
+
+    def highest_overlap(regionmask, intensity_image,
+                        overlap_threshold=intersection_area_over_object_area):
+        """
+        Gets the label number with highest overlap with label in another image.
+        
+        This function masks a labeled image called 'intensity_image' with 
+        'regionmask' and calculates the frequency of pixel values in that
+        region. Disconsidering zeros (background), it returns the most frequent
+        value if it overcomes the 'overlap_threshold'. Otherwise, it returns 0.
+        In case of draws, it returns the first occurence. This function follows
+        the standards of skimage.regionprops 'extra_properties'.
+         
+        Parameters
+        ----------
+        regionmask : (M, N[,P]) ndarray
+            Label image (probe channel). Labels to be used as a mask.
+        intensity_image : (M, N[,P]) ndarray
+            Label image (reference channel). Labels to be measured using probe
+            channel as a mask.
+        
+        Returns
+        -------
+        value : int
+            Most frequent label number under regionmask, except 0, that
+            overcomes threshold. Otherwise, it returns 0.
+        """
+        if overlap_threshold == 0:
+            return 0
+        values, counts = np.unique(np.sort(intensity_image[regionmask]),
```

Comment:
I don't understand this line. Please put a comment on top explaining why this is necessary.

Furthermore, is the `np.sort` really necessary? Isn't it computationally super expensive?

Reply:
I'm wondering if the functionality hidden between the lines here is similar to [`sklearn.metrics.confusion_matrix`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html)